src/libopensc/card-openpgp.c

```diff
@@ -650,6 +653,80 @@ pgp_parse_algo_attr_blob(sc_card_t *card, const pgp_blob_t *blob,
 	LOG_FUNC_RETURN(card->ctx, SC_SUCCESS);
 }
 
+int _pgp_handle_curve25519(sc_card_t *card,
+	sc_cardctl_openpgp_keygen_info_t key_info, size_t do_num)
+{
+	if (!sc_compare_oid(&key_info.u.ec.oid, &curve25519_oid))
+		return 0;
+
+	/* CKM_XEDDSA supports both Sign and Derive, but
+	* OpenPGP card supports only derivation using these
+	* keys as far as I know */
+	_sc_card_add_xeddsa_alg(card, key_info.u.ec.key_length,
+	    SC_ALGORITHM_ECDH_CDH_RAW, 0, &key_info.u.ec.oid);
+	sc_log(card->ctx, "DO %zX: Added XEDDSA algorithm (%d), mod_len = %d",
+	    do_num, SC_ALGORITHM_XEDDSA, key_info.u.ec.key_length);
+	return 1;
+}
+
+int _pgp_add_algo(sc_card_t *card, sc_cardctl_openpgp_keygen_info_t key_info, size_t do_num)
+{
+	unsigned long flags = 0, ext_flags = 0;
+
+	/* RSA [RFC 4880] */
```

Comment:
the comment looks misplaced

Reply:
It was like that before (https://github.com/OpenSC/OpenSC/pull/2287/files#diff-36989530cd06078fd8bce35f8f4c950abceca45e73980a5fddcd3731658db775L804), but I'll change it to mention https://tools.ietf.org/html/draft-ietf-openpgp-crypto-refresh